pkgs/development/compilers/zig/hook.nix

```
@@ -10,10 +10,13 @@ makeSetupHook {
 
   substitutions = {
     zig_default_flags =
-      if lib.versionAtLeast zig.version "0.11" then
-        "-Doptimize=ReleaseSafe -Dcpu=baseline"
-      else
-        "-Drelease-safe=true -Dcpu=baseline";
+      let
+        cpu = "-Dcpu=baseline";
+        releaseType = if lib.versionAtLeast zig.version "0.11"
+                      then "-Doptimize=ReleaseSafe"
+                      else "-Drelease-safe=true";
+      in
+        lib.concatStringsSep " " [ cpu releaseType ];
```

this is a noop, and why the unneccessary rebuilds?
```suggestion
      [ cpu releaseType ];
```

---

How can it be a mass rebuild if the result is the same?
And it was written as a string before, not as a list.